src/openxr/OpenXRApi.cpp

```diff
@@ -1999,90 +2067,95 @@ bool OpenXRApi::initialiseSwapChains() {
+	if (buffer_index == nullptr) {
+		Godot::print_error("OpenXR Couldn't allocate memory for buffer indexes", __FUNCTION__, __FILE__, __LINE__);
+		return false;
+	}
 
 	// Damn you microsoft for not supporting this!!
-	// uint32_t swapchainLength[view_count];
-	uint32_t *swapchainLength = (uint32_t *)malloc(sizeof(uint32_t) * view_count);
-	if (swapchainLength == nullptr) {
+	// uint32_t swapchain_length[swapchain_count];
+	uint32_t *swapchain_length = (uint32_t *)malloc(sizeof(uint32_t) * swapchain_count);
+	if (swapchain_length == nullptr) {
 		Godot::print_error("OpenXR Couldn't allocate memory for swap chain lengths", __FUNCTION__, __FILE__, __LINE__);
 		return false;
 	}
+	uint32_t *color_swapchain_length = swapchain_length;
+	uint32_t *depth_swapchain_length = swapchain_length + view_count;
 
 	for (uint32_t i = 0; i < view_count; i++) {
-		// again Microsoft wants these in order!
-		XrSwapchainCreateInfo swapchainCreateInfo = {
-			.type = XR_TYPE_SWAPCHAIN_CREATE_INFO,
-			.next = nullptr,
-			.createFlags = 0,
-			.usageFlags = XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_COLOR_ATTACHMENT_BIT,
-			.format = swapchain_format_to_use,
-			.sampleCount = swapchain_sample_count, // 1,
-			.width = render_target_width,
-			.height = render_target_height,
-			.faceCount = 1,
-			.arraySize = 1,
-			.mipCount = 1,
-		};
-
-		void **swapchain_create_info_pointer = const_cast<void **>(&swapchainCreateInfo.next);
-		for (XRExtensionWrapper *wrapper : registered_extension_wrappers) {
-			void **swapchain_create_info_next_pointer = wrapper->set_swapchain_create_info_and_get_next_pointer(swapchain_create_info_pointer);
-			if (*swapchain_create_info_pointer && swapchain_create_info_next_pointer && !*swapchain_create_info_next_pointer) {
-				swapchain_create_info_pointer = swapchain_create_info_next_pointer;
-			} else {
-				// Invalid return values.
-				// Reset the value stored by the swapchain_create_info_pointer so it can be reused in the next loop.
-				*swapchain_create_info_pointer = nullptr;
-			}
-		}
-
-		result = xrCreateSwapchain(session, &swapchainCreateInfo, &swapchains[i]);
-		if (!xr_result(result, "Failed to create swapchain {0}!", i)) {
-			free(swapchainLength);
+		// Create our color swapchain for our view first
+		if (!createSwapChain(
+					0,
+					XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_COLOR_ATTACHMENT_BIT,
+					swapchain_color_format_to_use,
+					swapchain_sample_count,
+					render_target_width,
+					render_target_height,
+					color_swapchains[i],
+					color_swapchain_length[i])) {
+			free(swapchain_length);
 			return false;
 		}
 
-		result = xrEnumerateSwapchainImages(swapchains[i], 0, &swapchainLength[i], nullptr);
-		if (!xr_result(result, "Failed to enumerate swapchains")) {
-			free(swapchainLength);
+		// now create our depth buffer for our view
+		if (!createSwapChain(
+					0,
+					XR_SWAPCHAIN_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT,
```

Comment:
The Meta's space warp sample uses this flag to create the depth texture swapchain:
`XR_SWAPCHAIN_USAGE_SAMPLED_BIT | XR_SWAPCHAIN_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT`

Reply:
For these flags you should only care about what you as app want to use them as, if the runtime needs to sample the textures/images internally then it will add the flags internally.

Reply:
I've added it in, indeed for OpenGL it is probably ignore but it will make it easier once I port the code to Godot 4.